mllib/src/main/scala/org/apache/spark/ml/classification/OneVsRest.scala

```diff
@@ -325,8 +350,13 @@ final class OneVsRest @Since("1.4.0") (
       multiclassLabeled.persist(StorageLevel.MEMORY_AND_DISK)
     }
 
+    val iters = Range(0, numClasses).par
+    iters.tasksupport = new ForkJoinTaskSupport(
+      new ForkJoinPool(Math.min(getParallelism, numClasses))
+    )
+
     // create k columns, one for each binary classifier.
-    val models = Range(0, numClasses).par.map { index =>
+    val models = iters.map { index =>
```

Comment:
Using `Futures` is a simple change and imho is clearer than using parallel collections once you start messing around with the task support.  It would just require changing a couple lines to something like this
```
  ...
  Future {classifier.fit(trainingDataset, paramMap)} (executionContext)
}.map(ThreadUtils.awaitResult(_, Duration.Inf)
```